### ISSUE TITLE
Improve `Style/Alias` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12351]
## Description

Always use `alias_method` when aliasing methods of modules, classes, or singleton classes at runtime, as the lexical scope of `alias` leads to unpredictability in these cases.

https://rubystyle.guide/#alias-method

## Describe the solution you'd like

```ruby
# bad

module Mononymous
  extend ActiveSupport::Concern

  included do
    delegate :my_method2, to: :class
  end

  module ClassMethods
    def my_method1; end

    alias my_method2 my_method1
  end
end

# good

module Mononymous
  extend ActiveSupport::Concern

  included do
    delegate :my_method2, to: :class
  end

  module ClassMethods
    def my_method1; end

    alias_method :my_method2, :my_method1
  end
end
```
